src/service.rs

```diff
@@ -226,15 +278,18 @@ where
 /// Start a node with the given parachain `Configuration` and relay chain `Configuration`.
 ///
 /// This is the actual implementation that is abstract over the executor and the runtime api.
+#[allow(clippy::too_many_arguments)]
```

Comment:
`start_node_impl` is a bit more complex now, mostly to handle the fact that the altair & centrifuge nodes aren't EVM nodes yet. A lot of this can be simplified back down to how it was once all the runtimes support EVM.

Reply:
Can't we have a command for running normal chain and one for running chain with EVM?

Reply:
The service file is by far the most complex piece to maintain at "parity" (no pun intended) with Polkadot/Substrate upgrades. If separating in two different implementations is doable, I would go with that route.

Reply:
My main concern with splitting into two different impls was exactly the maintenance cost - we'll need to update both `start_node_impl` versions whenever we do a substrate upgrade. But I also acknowledge that the code as written here makes the node startup flow even more complex than it used to be (and it was already pretty complex)
I'm happy to just create a second `start_evm_node_impl` that we can just migrate other runtimes to as they add EVM suport.

Reply:
@branan is there already some progress on splitting the two?